proposals/0000-add-support-for-display-p3-colors.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Add Support for Wide Gamut (DisplayP3) Colors to React Native
+author:
+  - Ryan Linton
+  - Yulian Glukhenko
+date: 2023-11-20
+---
+
+# RFC0000: Add Support for Wide Gamut (DisplayP3) Colors to React Native
+
+## Summary
+
+React Native does [not currently support](https://github.com/facebook/react-native/issues/41517) wide gamut color spaces (i.e. display-p3). This proposal discusses adding support for this color space in React Native, covering:
+
+- JS implementation
+- Platform changes
+
+## Basic example
+
+Add a DisplayP3 background color and text color using color() function syntax per the [W3C CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/#color-function) spec.
+
+### Using StyleSheet
+
+```js
+StyleSheet.create({
+  view: { backgroundColor: "color(display-p3 1 0.5 0)" },
+  text: { color: "color(display-p3 0 0.5 1)" },
+});
+```
+
+### Using inline styles
+
+```jsx
+const MyComp = () => (
+  <View style={{ backgroundColor: "color(display-p3 1 0.5 0)" }}>
+    <Text style={{ color: "color(display-p3 0 0.5 1)"  }}>
+  </View>
+)
+```
+
+### Using a feature flag
+
+Opt-in to using DisplayP3 as the default color space by using a feature flag.
+
+```sh
+# in ios/Podfile
+ENV['RCT_WIDE_GAMUT_ENABLED'] = '1'
+
+# in android/gradle.properties
+wideGamutEnabled=true
```

Comment:
I'd rather not use this build-time approach as it pollutes the codebase with checks all-over the place.

We can use feature flags like this one in the [`RCTBridge`](https://github.com/facebook/react-native/blob/main/packages/react-native/React/Base/RCTBridge.h#L45-L46).

In our case, they will not go in the RCTBridge file, of course, but this is a detail we can figure out later.

How it expect to work is that we put the creation of a color in its own factory function that takes the color parameters and the color space.
If the color space is not passed, it reads from the global feature flag what is the default space.
But we should try to avoid compile time flags as they are hard to read and they are hard to maintain.